contrib/gendocs.py

```diff
@@ -0,0 +1,297 @@
+* xref:doc/pages/scopes.adoc[Scopes]
+* xref:doc/pages/faces.adoc[Faces]
+* xref:doc/pages/buffers.adoc[Buffers]
+* xref:doc/pages/registers.adoc[Registers]
+* xref:doc/pages/mapping.adoc[Mapping]
+* xref:doc/pages/hooks.adoc[Hooks]
+* xref:doc/pages/command-parsing.adoc[Command Parsing]
+* xref:doc/pages/keys.adoc[Keys]
+* xref:doc/pages/regex.adoc[Regex]
+* xref:doc/pages/options.adoc[Options]
+* xref:doc/pages/highlighters.adoc[Highlighters]
+* xref:doc/pages/modes.adoc[Modes]
+* xref:doc/pages/keymap.adoc[KEYMAP]
+* xref:doc/pages/faq.adoc[FAQ]
+* xref:doc/pages/changelog.adoc[Changelog]
+* xref:doc/design.adoc[Design]
+* xref:doc/coding-style.adoc[Coding Style]
+* xref:doc/writing_scripts.adoc[Writing Scripts]
+* xref:doc/json_ui.adoc[JSON UI]
+* xref:doc/autoedit.adoc[Autoedit]
+* xref:doc/interfacing.adoc[Interfacing]
+* xref:rc/tools/lint.adoc[Linting]
+* xref:rc/tools/autorestore.adoc[Autorestore]
+* xref:rc/tools/doc.adoc[Doc]
+* xref:test/tests.adoc[Tests]
+* xref:VIMTOKAK.adoc[Vi(m) to Kakoune]
+"""
+
+with open("modules/ROOT/nav.adoc", "w") as f:
+    f.write(nav_content)
+
+# Antora component description file.
+# See: https://docs.antora.org/antora/latest/component-version-descriptor.
+antora_yml_content = """
+name: Kakoune
+nav:
+  - modules/ROOT/nav.adoc
+title: Kakoune
+version: latest
+"""
+
+with open("antora.yml", "w") as f:
+    f.write(antora_yml_content)
+
+# Antora playbook file.
+# See: https://docs.antora.org/antora/latest/playbook.
+antora_playbook_content = """
+asciidoc:
+  attributes:
+
+    # Do not complain on missing attributes,
+    # TODO: fix and turn to a fatal warning
+    attribute-missing: skip
+
+    # To fix links
+    idprefix: ""
+
+    # To fix links
+    idseparator: "-"
+
+    # Better to be reproducible, in general
+    reproducible: true
+
+    # More convenient to turn sections to IDs
+    sectids: true
+
+    # More convenient to have sections as links
+    sectlinks: true
+
+    # Do not want to miss something
+    sectnumlevels: 5
+
+    # More convenient to number the sections
+    sectnums: true
+
+    # Do not want to miss something
+    toclevels: 5
+
+  sourcemap: true
+
+content:
+  sources:
+  - url: ./
+    branches: ["HEAD"]
+
+runtime:
+  cache_dir: ./cache # More convenient for the development
+  fetch: true # More convenient for the development
+
+  log:
+    failure_level: fatal
+    level: warn
+
+output:
+  clean: true # More convenient for the development
+  dir: ./build # Simpler to have it explicit in code
+
+site:
+  title: Kakoune Docs
+
+ui:
+  bundle:
+    url: https://gitlab.com/antora/antora-ui-default/-/jobs/artifacts/HEAD/raw/build/ui-bundle.zip?job=bundle-stable
+"""
+
+with open("antora-playbook.yml", "w") as f:
+    f.write(antora_playbook_content)
+
+# Finally, generate the documentation,
+# results will be saved to the output directory
+# as specified in the `antora-playbook.yml`.
+subprocess.run(["antora", "generate", "antora-playbook.yml"])
```

Comment:
In addition to broken symlink, antora hangs indefinitely if there is a fifo in $PWD.
Also on a clean secondary worktree, it failed with

    [23:39:47.976] FATAL (antora): Local content source must be a git repository: /home/johannes/git/kakoune2 (url: ./)

so it doesn't recognize that .git is a gitlink.
I'm not sure why it even interacts with Git.

Reply:
@krobelus I'm not sure why it's been behaving like that,
but after keeping the work in the dedicated `doc_gen/`
I didn't have to remove the symlink,
so maybe this issue is gone now?

Could you verify it again, please?